corehq/apps/reports/static/reports/v2/js/datagrid/bindingHandlers.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * todo add docstring
+ */
+
+hqDefine('reports/v2/js/datagrid/bindingHandlers', [
+    'jquery',
+    'knockout',
+    'hqwebapp/js/atwho',
+    'atjs',
+    'caretjs',
+], function (
+    $,
+    ko,
+    atwho
+) {
+    'use strict';
+
+    ko.bindingHandlers.slugAutocomplete = {
+        init: function (element) {
+            var $element = $(element);
+            if (!$element.atwho) {
+                throw new Error("The typeahead binding requires Atwho.js and Caret.js");
+            }
+
+            atwho.init($element, {
+                atwhoOptions: {
+                    displayTpl: function (item) {
+                        if (item.case_type) {
+                            return '<li><span class="label label-default pull-right">${case_type}</span> ${name}</li>';
```

Comment:
this is referencing very specific data (`case_type`), but is in a generic looking binding handler (`slugAutocomplete`).

Reply:
true, but it does have a default option for when that's not the case. I am not particularly worried yet on making this more generic just yet.